filechooser-portal/FileChooserDialog.vala

```diff
@@ -355,6 +355,14 @@ public class Files.FileChooserDialog : Hdy.Window, Xdp.Request {
             stack.notify["visible-child"].connect (() => {
                 revealer.reveal_child = stack.visible_child_name == "search";
             });
+
+            if (action == Gtk.FileChooserAction.SAVE) {
+                Idle.add (() => {
+                    entry.set_placeholder_text (_("Enter new filename"));
+                    entry.grab_focus ();
+                    return Source.REMOVE;
+                });
+            }
```

Comment:
there's no need for the Idle callback, we can set the placeholder right bellow when we are configuring the entry for the first time, and grab the focus in the show() method.

Reply:
I thought I had tried that and it didn't work but I'll try again.

Reply:
> there's no need for the Idle callback, we can set the placeholder right bellow when we are configuring the entry for the first time, and grab the focus in the show() method.

I find that an Idle callback is still needed when `entry.grab_focus` is put in the `show ()` method (after `window.focus (Gdk.CURRENT_TIME);`).  Something is grabbing focus away again later otherwise.  I agree it would be good to get rid of the Idle if possible.

I am using Code "Save As" to test and restarting the `io.elementary.files.xdg-desktop-portal` process between changes.

Reply:
i tested it and it was the TreeView re-grabbing the focus.

in the `current_folder_changed()` signal, we grab the focus to the TreeView to it don't end up in the LocationBar. we can grab the focus to the entry there too, but we will be focusing the entry always when the user change the current folder.

Reply:
That could be inconvenient for keyboard-only operation I guess but shouldn't be a problem for someone using pointer operation.  I do not know how many keyboard-only users there are or how to prioritise them?

@danirabbit Do you have any input on this, please?

Reply:
I'm not sure why we had the treeview grabbing focus every time the view changes. AFAICT, keyboard navigation in the view works as expected without having to re-focus.

I think it makes sense to have the view grab focus on `show ()` and then not try to be clever about moving focus around afterwards and let Gtk do the right thing from there